src/helics/shared_api_library/ValueFederateExport.cpp

```diff
@@ -455,10 +508,14 @@ HelicsInput helicsFederateGetInputByTarget(HelicsFederate fed, const char* targe
             assignError(err, HELICS_ERROR_INVALID_ARGUMENT, invalidTargetKey);
             return nullptr;
         }
-        auto inp = std::make_unique<helics::InputObject>();
-        inp->inputPtr = &id;
-        inp->fedptr = std::move(fedObj);
-        return addInput(fed, std::move(inp));
+        auto hInp = findInput(fed, id.getHandle());
+        if (hInp == nullptr) {
+            auto inp = std::make_unique<helics::InputObject>();
+            inp->inputPtr = &id;
+            inp->fedptr = std::move(fedObj);
+            hInp = addInput(fed, std::move(inp));
+        }
+        return hInp;
```

Comment:
This pattern feels like it could be its own function, or added as part of the `add*` or `find*` functions if it is a check that always needs to happen (`findOrCreate*?`).

Reply:
I submitted an update with this suggestion.